extract-variables/README.md

```diff
@@ -0,0 +1,105 @@
+# Extract Variables and Assign Message
+
+This sample shows how to easily extract fields from an XML and JSON response. For this sample I will be leveraging the following policies:
+
+ * [Extract Variables](https://cloud.google.com/apigee/docs/api-platform/reference/policies/extract-variables-policy) policy to extract subsets of data from the response body.
+ * [XML to JSON](https://cloud.google.com/apigee/docs/api-platform/reference/policies/xml-json-policy) policy to convert the XML response from the target server to JSON.
+ * [Assign Message](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy?hl=en) policy to set the response Headers with the extracted fields and other flow variables.
+
+## About the extract variables policy
+
+The ExtractVariables policy extracts content from a request or response and sets the value of a variable to that content. You can extract any part of the message, including headers, URI paths, JSON/XML payloads, form parameters, and query parameters. The policy works by applying a text pattern to the message content and, upon finding a match, sets a variable with the specified message content.
+
+While you often use ExtractVariables to extract information from a request or response message, you can also use it to extract information from other sources, including entities created by the AccessEntity policy, XML objects, or JSON objects.
+
+After extracting the specified message content, you can reference the variable in other policies as part of processing a request and response.
```

Comment:
Perhaps add a link to more info about flow variables and point to [this doc](https://cloud.google.com/apigee/docs/api-platform/fundamentals/introduction-flow-variables).